pkg/framework/resource/resource.go

```diff
@@ -34,7 +35,22 @@ func New(config ResourceConfig) (*Resource, error) {
 		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
 	}
 	if config.ApprClient == nil {
-		return nil, microerror.Maskf(invalidConfigError, "%T.ApprClient must not be empty", config)
+		config.Logger.Log("level", "warning", "message", fmt.Sprintf("%T.ApprClient is empty, using default", config))
```

Comment:
I think this should just be debug. I also think when you need a comma in a message you should emit two logs.

Reply:
Changed to debug, do you mean two logs like this? Or did I misunderstand you

Reply:
Yes I think this is better.